stream/pubsub.h

```diff
@@ -335,6 +340,89 @@ class PubSubHTTPEndpointImpl : public AbstractSubscriberObject {
     return result;
   }
 
+  ss::EntryResponse operator()(const std::string& entry_json, uint64_t current_index, idxts_t last) {
+    const ss::EntryResponse result = [&, this]() {
+      if (time_to_terminate_) {
+        return ss::EntryResponse::Done;
+      }
+      auto current_us = std::chrono::microseconds(0);
+      // Obtain current timestamp only when it's necessary by parsing the `entry_json`.
+      const auto GetCurrentUs = [this, &current_us, &entry_json]() -> std::chrono::microseconds {
+        if (!current_us.count()) {
+          current_us = ParseJSON<idxts_t>(entry_json.substr(0, entry_json.find('\t'))).us;
+        }
+        return current_us;
+      };
+      if (!serving_) {
+        if (current_index >= params_.i &&                                            // Respect `i`.
+            (params_.tail == 0u || (last.index - current_index) < params_.tail) &&   // Respect `tail`.
+            (from_timestamp_.count() == 0u || GetCurrentUs() >= from_timestamp_)) {  // Respect `since` and `recent`.
+          serving_ = true;
+        }
+        // Reached the end, didn't started serving and should not wait.
+        if (!serving_ && current_index == last.index && params_.no_wait) {
```

Comment:
This whole logic is almost identical to the `operator()` for the "safe" subscription case above.
1. `if (!serving_)` makes sense if the condition right above this line just became true and switched the `serving_` to `true`.
2. `current_index >= params_.i`, I suppose, covers the case when for some reason the first entry which is passed here has greater index then required to start serving.
3. `current_index` and `last.index` both passed from outside. So how can the first one possibly be greater then the second one?